client/ayon_core/hosts/resolve/plugins/load/load_media.py

```diff
@@ -0,0 +1,346 @@
+        media_pool = project.GetMediaPool()
+
+        # Allow to use an existing media pool item and re-use it
+        item = None
+        if options.get("load_once", True):
+            host = registered_host()
+            repre_id = str(context["representation"]["_id"])
+            for container in host.ls():
+                if container["representation"] != repre_id:
+                    continue
+
+                if container["loader"] != self.__class__.__name__:
+                    continue
+
+                print(f"Re-using existing container: {container}")
+                item = container["_item"]
+
+        if item is None:
+            # Create or set the bin folder, we add it in there
+            # If bin path is not set we just add into the current active bin
+            if self.media_pool_bin_path:
+                media_pool_bin_path = StringTemplate(
+                    self.media_pool_bin_path).format_strict(context)
+                folder = lib.create_bin(
+                    name=media_pool_bin_path,
+                    root=media_pool.GetRootFolder(),
+                    set_as_current=False
+                )
+                media_pool.SetCurrentFolder(folder)
+
+            # Import media
+            path = self._get_filepath(representation)
+            items = media_pool.ImportMedia([path])
+
+            assert len(items) == 1, "Must import only one media item"
+            item = items[0]
+
+            self._set_metadata(media_pool_item=item,
+                               context=context)
+
+            data = self._get_container_data(representation)
+
+            # Add containerise data only needed on first load
+            data.update({
+                "schema": "openpype:container-2.0",
+                "id": AVALON_CONTAINER_ID,
+                "loader": str(self.__class__.__name__),
+            })
+
+            item.SetMetadata(lib.pype_tag_name, json.dumps(data))
+
+        # Always update clip color - even if re-using existing clip
+        color = self.get_item_color(representation)
+        item.SetClipColor(color)
+
+        if options.get("load_to_timeline", True):
+            timeline = options.get("timeline", self.timeline)
+            if timeline:
+                # Add media to active timeline
+                lib.create_timeline_item(
+                    media_pool_item=item,
+                    timeline=timeline
+                )
+
+    def switch(self, container, representation):
+        self.update(container, representation)
+
+    def update(self, container, representation):
+        # Update MediaPoolItem filepath and metadata
+        item = container["_item"]
+
+        # Get the existing metadata before we update because the
+        # metadata gets removed
+        data = json.loads(item.GetMetadata(lib.pype_tag_name))
+
+        # Update path
+        path = self._get_filepath(representation)
+        item.ReplaceClip(path)
```

Comment:
🔴 

Just found a bug in this code in our production. This doesn't actually work.

Even though the documentation states this can take a filepath, it actually doesn't - it takes a folder path and will remap the MediaPoolItem with a clip in that new folder. And it looks like it will just take the FIRST media file found in that folder, instead of the one we actually want.

I'm looking into other options.

Reply:
This fixed it for us: https://github.com/BigRoy/OpenPype/commit/e22d5dbb9fe5b9006a6b81bb156e875c47ab3638

The issue was only with sequences.